src/ripple/app/misc/detail/WorkSSL.h

```diff
@@ -102,24 +77,24 @@ WorkSSL::WorkSSL(
     std::string const& port,
     boost::asio::io_service& ios,
     beast::Journal j,
+    Config const& config,
     callback_type cb)
     : WorkBase(host, path, port, ios, cb)
-    , context_(j)
-    , stream_(socket_, context_)
+    , context_(config, j)
+    , stream_(socket_, context_.context())
 {
-    // Set SNI hostname
-    SSL_set_tlsext_host_name(stream_.native_handle(), host.c_str());
-    stream_.set_verify_mode (boost::asio::ssl::verify_peer);
-    stream_.set_verify_callback(    std::bind (
-            &WorkSSL::rfc2818_verify, host_,
-            std::placeholders::_1, std::placeholders::_2));
+    auto ec = context_.preConnectVerify(stream_, host_);
+    if (ec)
+        Throw<std::runtime_error> (
+            boost::str (boost::format ("preConnectVerify: %s") % ec.message ()));
 }
 
 void
 WorkSSL::onConnect(error_code const& ec)
 {
-    if (ec)
-        return fail(ec);
+    auto err = ec ? ec : context_.postConnectVerify(stream_, host_);
+    if (err)
+        return fail(err);
```

Comment:
Outside the scope of this PR but we should be using the new stream timer expiration here.

Reply:
a timer for `postConnectVerify`, or just generally for all stream/async operations ? if the latter, I agree...if the former, I guess I need more clarification.

Reply:
for all stream/async operations.

Reply:
agree..will make an issue for this.